### PR TITLE
Features/7986

### PR DESCRIPTION
**Pull request by FuriousLlama**
## Status
**Ready**

## Fixes or Implements [AB#7986](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/7986)
- [x] Enhancement
- [ ] Bug
- [ ] Documentation

## Description:
Controller , service and model to return data from the ODR delegate

## Testing
- [X] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

### UI Changes
NO

### CODE REVIEW

File: Apps/Medication/src/Controllers/MedicationStatementController.cs [hunk 1]
```diff
@@ -20,19 +20,21 @@ namespace HealthGateway.Medication.Controllers
...
+    using Microsoft.Extensions.Configuration;
 
     /// <summary>
     /// The Medication controller.
     /// </summary>
     [ApiVersion("1.0")]
     [Route("v{version:apiVersion}/api/[controller]")]
     [ApiController]
-    [Authorize]
+    //[Authorize]
```
Comment (tiagograf-AOT) [inline Apps/Medication/src/Controllers/MedicationStatementController.cs]: did you forget to uncomment this?

File: Apps/Medication/src/Controllers/MedicationStatementController.cs [hunk 2]
```diff
@@ -75,32 +84,32 @@ public MedicationStatementController(IAuthorizationService authorizationService,
         [HttpGet]
         [Produces("application/json")]
         [Route("{hdid}")]
-        [Authorize(Policy = "PatientOnly")]
+        //[Authorize(Policy = "PatientOnly")]
```
Comment (tiagograf-AOT) [inline Apps/Medication/src/Controllers/MedicationStatementController.cs]: same

File: Apps/Medication/src/Controllers/MedicationStatementController.cs [hunk 3]
```diff
@@ -75,32 +84,32 @@ public MedicationStatementController(IAuthorizationService authorizationService,
...
         public async Task<IActionResult> GetMedicationStatements(string hdid, [FromHeader] string? protectiveWord = null)
         {
-            ClaimsPrincipal user = this.httpContextAccessor.HttpContext.User;
+            /*ClaimsPrincipal user = this.httpContextAccessor.HttpContext.User;
```
Comment (tiagograf-AOT) [inline Apps/Medication/src/Controllers/MedicationStatementController.cs]: same

File: Apps/Medication/src/Controllers/MedicationStatementController.cs (hunk 1 above)
Comment (FuriousLlama) [reply Apps/Medication/src/Controllers/MedicationStatementController.cs]: Fixed

File: Apps/Medication/src/Controllers/MedicationStatementController.cs (hunk 3 above)
Comment (FuriousLlama) [reply Apps/Medication/src/Controllers/MedicationStatementController.cs]: Fixed

File: Apps/Medication/src/Controllers/MedicationStatementController.cs (hunk 2 above)
Comment (FuriousLlama) [reply Apps/Medication/src/Controllers/MedicationStatementController.cs]: Fixed

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_392)

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_392)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_392)

**Pull request merged**